lib/charms/oathkeeper/v0/auth_proxy.py

```diff
@@ -337,6 +342,33 @@ def _on_relation_departed_event(self, event: RelationDepartedEvent) -> None:
         """Notify Oathkeeper that the relation has departed."""
         self.on.config_removed.emit(event.relation.id)
 
+    def get_headers(self) -> Optional[List[str]]:
+        """Returns the list of headers from all relations."""
+        if len(self.model.relations) == 0:
+            return None
+
+        headers = list()
+        for relation in self._charm.model.relations[self._relation_name]:
+            if not relation.data[relation.app]:
+                return None
+            for header in json.loads(relation.data[relation.app]["headers"]):
+                # Avoid duplicates
+                if header not in headers:
+                    headers.append(header)
+
+        return headers
+
+    def get_app_names(self) -> Optional[List[str]]:
+        """Returns the list of all related app names."""
+        if len(self.model.relations) == 0:
+            return None
```

Comment:
Is this check needed? You care about `self._charm.model.relations[self._relation_name]`, not all the relations.

Also maybe you would want to return `[]` instead of `None` because you have defined app_names to be not optional (see https://github.com/canonical/oathkeeper-operator/pull/36/files#diff-939758562ec99f096aa02966a5483b056e281b4b8be021b745715357ce664cfaR174)